gammapy/maps/hpx.py

```diff
@@ -1731,6 +1732,23 @@ def solid_angle(self):
         import healpy as hp
         return Quantity(hp.nside2pixarea(self.nside), 'sr')
 
+    def get_region_idx(self, region):
+        """Return idx of pixels inside region
+
+        TODO: implement list of region for each axis
+
+        Parameters
+        ----------
+        region : `~regions.PixelRegion` or `~regions.SkyRegion` object
+            A region on the sky could be defined in pixel or sky coordinates.
+        """
+        if not isinstance(region, SkyRegion):
+            raise NotImplementedError
+
+        coords=self.get_coord()
+        return region.contains(coords.skycoord)
```

Comment:
Note that in future ``regions`` releases ``contains`` will require a ``wcs`` object (see [latest docs](http://astropy-regions.readthedocs.io/en/latest/api/regions.SkyRegion.html?#regions.SkyRegion.contains)). This is due to the fact, that ``SkyRegions`` are basically ``PixelRegions`` with the coordinates defined on the sky (as opposed to real sky regions). 

So this method should work for ``PixelRegions`` - at the moment you raise a ``NotImplementedError`` on that. For convenience you could accept ``SkyRegion`` plus a ``wcs`` object. But like it is now this method will stop working soon.